cmd/mavenExecute.go

```diff
@@ -0,0 +1,110 @@
+package cmd
+
+import (
+	"bytes"
+	"github.com/SAP/jenkins-library/pkg/http"
+	"io"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+const mavenExecutable = "mvn"
+
+func mavenExecute(config mavenExecuteOptions, telemetryData *telemetry.CustomData) string {
+	c := command.Command{}
+	output, err := runMavenExecute(&config, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+	return output
+}
+
+func runMavenExecute(config *mavenExecuteOptions, command execRunner) (string, error) {
+	stdOutBuf, stdOut := evaluateStdOut(config)
+	command.Stdout(stdOut)
+	command.Stderr(log.Entry().Writer())
+
+	parameters := getParametersFromConfig(config, &http.Client{})
+
+	err := command.RunExecutable(mavenExecutable, parameters...)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			WithField("command", append([]string{mavenExecutable}, parameters...)).
+			Fatal("failed to execute run command")
+	}
+
+	if stdOutBuf == nil {
+		return "", nil
+	}
+	return string(stdOutBuf.Bytes()), nil
+}
+
+func evaluateStdOut(config *mavenExecuteOptions) (*bytes.Buffer, io.Writer) {
+	var stdOutBuf *bytes.Buffer
+	var stdOut io.Writer
+
+	stdOut = log.Entry().Writer()
+	if config.ReturnStdout {
+		stdOutBuf = new(bytes.Buffer)
+		stdOut = io.MultiWriter(stdOut, stdOutBuf)
+	}
+	return stdOutBuf, stdOut
+}
+
+func getParametersFromConfig(config *mavenExecuteOptions, client http.Downloader) []string {
+	var parameters []string
+
+	if config.GlobalSettingsFile != "" {
+		globalSettingsFileParameter := "--global-settings " + config.GlobalSettingsFile
+		if strings.HasPrefix(config.GlobalSettingsFile, "http") {
```

Comment:
I would like to propose to check for `http:` and for `https:`. Otherwise the string `httpxxxxx` would also be recognized as valid protocol. In case we use shared coding here and for the mta build we can control that in one place.